ZiLing/Midterm-Review.py

```diff
@@ -0,0 +1,481 @@
+print(2%2)
+print(3%2)
+
+# be aware of the difference in decimal places
+print(2//3)
+print(2/3)
+
+# you can also use + and * on string 
+word = "hello" + " world"
+hohoho = 'ho'*3
+
+"""
+Assignment Operators (in-place operators)
+You dont have to use them, but just know what they are 
+
+= -> x = 5 
++= -> x += 3 is the same sa x = x + 3 
+-= -> x -= 3 is the same as x = x - 3 
+*= -> x *= 3 is the same sa x = x * 3 
+/= -> x /= 3 is the same sa x = x / 3 
+%= -> x %= 3 is the same sa x = x % 3 
+**= -> x **= 3 is the same sa x = x ** 3 
+
+there are more than these, but just know the onces above should be enough 
+for you to go through this course
+
+"""
+
+# You cannot use two assignment operators on the same variable at the same line 
+# But you can mix it with other algebretical operators 
+x += 435+493/2302567
+
+"""
+Everything below will return a boolean value(True or False)
+
+Comparison Oprators 
+
+Operator   Name           Example
+==         Equal          x == y, 2 == 1 
+!=       Not Equal       x != 7, y != y
+>      Greater than       3 > 2, 4 > 5
+<       Less than         1 < 2, 5 < 5 
+>=   Greater or Equal to   1 >= 1, 1 >= 4
+<=   less or equal to      x <= 3 , 4 <= 4 
+
+You usually use them with if statement or while statement
+"""
+
+# You can use more than one comparison operators in the same line 
+for i in range (356,9,-3): 
+    if 325 < i > 35:
+        print (i)
+
+"""
+Logical Oprator 
+
+operator    Returns True if          Example
+and     both statements are true     x < 5 and x > 1 
+or      either statement is true     x < 5 or x > 1 
+not     the statement is false       not(x > 5)
+
+Identity Operator 
+
+operator    Returns True if          Example
+is      both variables are the same   5 is 5, x is 5
+is not  variables are not the same    x is not 4 
+
+Membership Operator 
+
+operator    Returns True if                 Example
+in       object is element in smt        i in mydict
+not in      object not in smth           i not in mydict     
+
+All the operators above return a boolean type value (True or False)
+"""
+
+# You can mix mash all the operators together as long as it makes sense
+for i in range (1,101):
+    if not(x%2 == 0 and x > 5 ): 
+        print (i)
+
+"""
+Conditional statements 
+
+Statement      name   
+if              if      
+elif         else if
+else           else
+while          while 
+"""
+
+"""
+Range
+
+range is a type that gives a range of intergers, like how 
+string is a sequence of characters, range is a sequence 
+of integers
+
+range(start,end,number between each)
+"""
+# This will not print out every value in range
+print(range(4))
+
+# This will
+for i in range(4):
+    print(i)
+
+#After talking about math and boolean, lets talk a little about string 
+
+"""
+String 
+
+String is a sequence of characters, or rather, a list of characters,
+meaning all functions that works with list will work with string 
+in the same way, for example 
+"""
```

Comment:
Not quite. A string is implemented as an **array** of chars are the C level. It does not use python lists.